jetbrains-rider/tst/software/aws/toolkits/jetbrains/services/clouddebug/DotNetStartupCommandAugmenterTest.kt

```diff
@@ -0,0 +1,140 @@
+import org.testng.annotations.AfterMethod
+import org.testng.annotations.BeforeMethod
+import org.testng.annotations.Test
+import software.aws.toolkits.resources.message
+
+class DotNetStartupCommandAugmenterTest : AwsReuseSolutionTestBase() {
+
+    override fun getSolutionDirectoryName() = "SamHelloWorldApp"
+
+    companion object {
+        private const val DEFAULT_STARTUP_COMMAND = "dotnet /prog/netcoreapp2.1/ConsoleApp.dll"
+    }
+
+    var useNetCoreDebuggerOriginal: Boolean = true
+
+    @BeforeMethod(alwaysRun = true)
+    fun setRegistry() {
+        useNetCoreDebuggerOriginal = Registry.get(DotNetDebuggerSupport.USE_DOTNET_CORE_RUNTIME_FLAG_NAME).asBoolean()
+    }
+
+    @AfterMethod(alwaysRun = true)
+    fun resetRegistry() {
+        Registry.get(DotNetDebuggerSupport.USE_DOTNET_CORE_RUNTIME_FLAG_NAME).setValue(useNetCoreDebuggerOriginal)
+    }
+
+    @Test
+    fun testAugmentStatement_NoDebugPort_Exception() {
+        assertThatThrownBy { DotNetDebuggerSupport().augmentStatement(DEFAULT_STARTUP_COMMAND, listOf(), "") }
+            .isInstanceOf(IllegalStateException::class.java)
+            .hasMessage(message("cloud_debug.step.augment_statement.missing_debug_port"))
+    }
+
+    @Test
+    fun testAugmentStatement_SingleDebugPort_Exception() {
+        assertThatThrownBy { DotNetDebuggerSupport().augmentStatement(DEFAULT_STARTUP_COMMAND, listOf(123), "") }
+            .isInstanceOf(IllegalStateException::class.java)
+            .hasMessage(message("cloud_debug.step.dotnet.two_ports_required"))
+    }
+
+    @Test
+    fun testAugmentStatement_MonoRuntime_ChmodToRuntimeAndMono() {
+        Registry.get(DotNetDebuggerSupport.USE_DOTNET_CORE_RUNTIME_FLAG_NAME).setValue(false)
+
+        val pathToDebugger = "/path/to/debugger"
+        val statement = DotNetDebuggerSupport().augmentStatement(DEFAULT_STARTUP_COMMAND, listOf(123, 456), pathToDebugger)
+        val expectedCommand =
+            ParametersListUtil.join(
+                "/aws/cloud-debug/common/busybox",
+                "sh",
+                "-c",
+                ParametersListUtil.join(
+                    "/aws/cloud-debug/common/busybox",
+                    "chmod",
+                    "+x",
+                    "/aws/DOTNET/aws_rider_debugger_files/runtime.sh",
+                    "/aws/DOTNET/aws_rider_debugger_files/linux-x64/mono/bin/mono-sgen",
+                    "&&",
+                    "env",
+                    "REMOTE_DEBUG_PORT=123",
+                    "RESHARPER_HOST_LOG_DIR=/aws/DOTNET/aws_rider_debugger_files/Logs",
+                    "/aws/DOTNET/aws_rider_debugger_files/runtime.sh",
+                    pathToDebugger,
+                    "--mode=server",
+                    "--frontend-port=123",
+                    "--backend-port=456"
+                )
+            )
+        assertThat(statement).isEqualTo(expectedCommand)
+    }
+
+    @Test
+    fun testAugmentStatement_DotNetCoreRuntime_ChmodToRuntimeAndMono() {
```

Comment:
`_ChmodToRuntimeAndMono` part doesnt seem to match test

Reply:
Re-named.